channels/rdpdr/client/rdpdr_main.c

```diff
@@ -495,14 +500,22 @@ static UINT handle_hotplug(rdpdrPlugin* rdpdr)
 				free(drive->Path);
 				free(drive->Name);
 				free(drive);
-				return error;
+				error = CHANNEL_RC_NO_MEMORY;
+				goto cleanup;
 			}
 		}
-
-		free(dev_array[i].path);
-		dev_array[i].path = NULL;
 	}
+
+	for (i = 0; i < size; i++)
+		free (dev_array[size].path);
+
 	return rdpdr_send_device_list_announce_request(rdpdr, TRUE);
+
+cleanup:
+	for (i = 0; i < size; i++)
+		free (dev_array[size].path);
+
+	return error;
```

Comment:
would nt it be better to remove the duplicated code above and use something like:

return error ? error : rdpdr_send_device_list_announce_request(rdpdr, TRUE);